render/render.js

```diff
@@ -775,38 +775,43 @@ module.exports = function($window) {
 		) && key in vnode.dom
 	}
 
-	var matchUpperCase = /[A-Z]/g
-	function prependDashAndLowerCase(string){
-		return "-" + string.toLowerCase()
-	}
-
-	function normalizeProp(prop) {
-		return "-" && prop[1] === "-"
-			? prop
-			: prop.replace(matchUpperCase, prependDashAndLowerCase)
-	}
-
 	//style
+	var uppercaseRegex = /[A-Z]/g
+	function toLowerCase(capital) { return "-" + capital.toLowerCase() }
+	function normalizeKey(key) {
+		return key[0] === "-" && key[1] === "-" ? key :
+			key === "cssFloat" ? "float" :
+				key.replace(uppercaseRegex, toLowerCase)
+	}
 	function updateStyle(element, old, style) {
-		if (old != null && style != null && typeof old === "object" && typeof style === "object" && style !== old) {
+		if (old === style) {
+			// Do nothing.
+		} else if (style == null) {
+			element.style.cssText = ""
+		} else if (typeof style !== "object") {
+			element.style.cssText = style
+		} else if (old == null || typeof old !== "object") {
+			element.style.cssText = ""
```

Comment:
Position of this line confused me first. Maybe move below the comment.

Reply:
Fixed. I also need to add a few more comments around that area.

Reply:
*needed